### PR TITLE
Hide add role if user is not entitled for cost management

### PR DESCRIPTION
**Pull request by karelhala**
### Changes

There's been a lot of redundant code and a lot of reundant calls to API. I've removed the code and streamlined role fetching. When user is not entitled to cost management no button will be shown. Also filter was not properly working, so I updated the fetch function to actually filter. Next step should be ordering and refactoring of groups table.

### UI

Screenshots taken on account without `cost_management` entitlements.

**Before**
Screenshot from 2019-11-06 16-42-20

**After - user not org admin or not on beta**
Screenshot from 2019-11-08 10-45-32

**After - user is org admin and on beta**
Screenshot from 2019-11-08 10-45-09

### CODE REVIEW

**Comment by Hyperkid123** [comment]
@karelhala just fix the test please.

**Comment by gmcculloug** [comment]
> When user is not entitled to cost management no button will be shown.

@karelhala Will this change affect Catalog usage?  Can you add a screenshot to clarify what is being changed?

**Comment by karelhala** [comment]
@gmcculloug yup, that would effect Catalog as well, we have a JIRA RHCLOUD-2932 to disable custom roles for users not entitled to cost-management. Let me upload the pictures.

**Comment by gmcculloug** [comment]
@karelhala Realizing this is not an issue for Catalog as we are not using custom roles.  👍

File: package.json [hunk 1]
```diff
@@ -55,6 +55,7 @@
     "@babel/plugin-transform-runtime": "^7.2.0",
     "@babel/preset-env": "^7.2.3",
     "@babel/preset-react": "^7.0.0",
+    "axios-mock-adapter": "^1.17.0",
```
Comment (Hyperkid123) [inline package.json]: I think you can remove the `xhr-mock` from the list.

**Pull request merged**